client/parser/scanner.go

```diff
@@ -97,10 +97,10 @@ const (
 
 // A Token is a result of lexical analysis done by the scanner.
 type Token struct {
-	sourceContext model.AldaSourceContext
-	tokenType     TokenType
-	text          string
-	literal       interface{}
+	SourceContext model.AldaSourceContext
+	TokenType     TokenType
+	Text          string
+	Literal       interface{}
```

Comment:
I think once we do the refactor to have the parser produce an AST, we might not need these to be public. I'm thinking that tokens would remain an internal implementation detail of the parser, which turns text -> tokens -> AST. I don't think anything outside of the parser will be concerned with tokens, so we should keep them private if possible.

Reply:
Right so if we use an AST representation, then the model/code generator packages where the code gen logic lies won't have to touch the actual token representation. 

That is likely a change we will make. Good point.